### DISCUSSION TITLE
tu_fifo_set_overwritable not being set back to false after Windows 10 sleep/wake

### DISCUSSION THREAD

**Discussion by belchercw** [hathach/tinyusb#738] [Q&A]
This may be due to my own lack of understanding, but I have found a problem with USB CDC serial transmission from device to host after the host has gone to sleep and then resumed (Host is WIndows 10 PC).  My device sends a steady stream of traffic to a custom application on the host.  My device code is careful to check the return value of tud_cdc_write, and if not all data has been written, to wait for tud_cdc_tx_complete_cb to proceed.  If I set my block size lower than CFG_TUD_CDC_TX_BUFSIZE, then when the PC resumes from sleep, the PC application continues to receive data from my device as usual.  If my block size is larger than CFG_TUD_CDC_TX_BUFSIZE, then before sleep things work great because the tx fifo is set to non-overwritable, but after resume from sleep, the tx fifo becomes overwritable, and thus tud_cdc_write discards data when passed these large buffers.  Perhaps I need to be explicitly handling the resume callback and doing something that will set overrwritable back to false?

**Comment by hathach** [comment]
- firstly could you provide context of your setup using format from the bug template (host os, board, mcu etc ..). 
- Then try to enable debug LOG to see what is going on inside the stack. 
- If nothing is suspicious from the log, try to come up with an reproducible/simplest test so that other can try it out

**Reply by belchercw** [reply]
Thank you for the prompt reply.  I have entered a bug report here  https://github.com/hathach/tinyusb/issues/743
including minimal changes to the cdc_msc demo project and instructions to reproduce.
